File: master-node-docker/sentinel/helpers/referral.py
# coding=utf-8
import requests

from ..config import REFERRAL_URL
from ..db import db


def add_bonus(device_id):
    try:
        url = '{}/accounts/{}/bonuses'.format(REFERRAL_URL, device_id)
        res = requests.post(url, timeout=15)
        res = res.json()
        return None, res['success']
    except Exception as error:
        return error, None


def get_vpn_sessions(device_id):
    usage = db.ref_sessions.find({
        'device_id': device_id
    }, {
        '_id': 0,
        'session_id': 0,
        'to_addr': 0,
    })
    return list(usage) if usage else []

File: master-node-nodejs/src/helpers/referral.js
import axios from "axios";
import {
  REFERRAL_URL
} from "../config/referral";

const addSession = (deviceId, sessionId, txHash = null, cb) => {
  try {
    let body = {
      'deviceId': deviceId,
      'session': {
        'sessionId': sessionId
      }
    }
    if (txHash) {
      body['session']['paymentTxHash'] = txHash
    }
    let url = `https://${REFERRAL_URL}/session`
    axios.post(url, body, {
      timeout: 10
    }).then((resp) => {
      resp = resp.data
      cb(null, resp['success'])
    })
  } catch (error) {
    cb(error, null)
  }
}

export {
  addSession
}
